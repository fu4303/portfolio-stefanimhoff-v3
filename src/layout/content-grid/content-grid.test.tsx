import React from 'react';

import { render } from '../../services/test-utils';

import { ContentGrid } from '.';

describe('Grid', () => {
  test('renders correctly', () => {
    const { container } = render(<ContentGrid>ContentGrid</ContentGrid>);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('with "fullsize" renders correctly', () => {
    const { container } = render(
      <ContentGrid size="fullsize">ContentGrid</ContentGrid>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('with "regular" renders correctly', () => {
    const { container } = render(
      <ContentGrid size="regular">ContentGrid</ContentGrid>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('with "narrow" renders correctly', () => {
    const { container } = render(
      <ContentGrid size="narrow">ContentGrid</ContentGrid>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('with "rowGap" renders correctly', () => {
    const { container } = render(<ContentGrid rowGap>ContentGrid</ContentGrid>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
