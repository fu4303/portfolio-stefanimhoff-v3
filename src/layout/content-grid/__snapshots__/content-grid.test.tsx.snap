// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
.c0 {
  grid-template-columns: repeat(18,1fr);
  display: grid;
  width: 100%;
}

<div
  class="c0"
>
  ContentGrid
</div>
`;

exports[`Grid with "fullsize" renders correctly 1`] = `
.c0 {
  grid-template-columns: repeat(18,1fr);
  display: grid;
  width: 100%;
}

<div
  class="c0"
>
  ContentGrid
</div>
`;

exports[`Grid with "narrow" renders correctly 1`] = `
.c0 {
  grid-template-columns: repeat(16,1fr);
  display: grid;
  width: 100%;
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(8,1fr);
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-template-columns: repeat(6,1fr);
  }
}

<div
  class="c0"
>
  ContentGrid
</div>
`;

exports[`Grid with "regular" renders correctly 1`] = `
.c0 {
  grid-template-columns: repeat(16,1fr);
  display: grid;
  width: 100%;
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(14,1fr);
  }
}

<div
  class="c0"
>
  ContentGrid
</div>
`;

exports[`Grid with "rowGap" renders correctly 1`] = `
.c0 {
  grid-template-columns: repeat(18,1fr);
  row-gap: clamp(25px,5.55vw,100px);
  display: grid;
  width: 100%;
}

<div
  class="c0"
>
  ContentGrid
</div>
`;
